File: frontend/src/types/printer.ts
export type PrinterState = "IDLE" | "PRINTING" | "STOPPED" | "ERROR" | "UNKNOWN";

export interface PrinterSummary {
  name: string;
  description: string;
  state: PrinterState;
  state_message: string;
  accepting_jobs: boolean;
  shared: boolean;
  device_uri: string | null;
  current_job: number | null;
  queue_count: number;
}

export interface PrinterCapabilities {
  media: string[];
  color: boolean;
  duplex: boolean;
  resolution: string[];
  copies_supported: boolean;
  max_copies: number | null;
  page_ranges_supported: boolean;
  orientation_supported: boolean;
}

export interface PrinterDetail extends PrinterSummary {
  location: string | null;
  manufacturer: string | null;
  model: string | null;
  capabilities: PrinterCapabilities;
}

File: backend/app/schemas/printer.py
from typing import Optional

from pydantic import BaseModel


class PrinterCapabilities(BaseModel):
    media: list[str] = []
    color: bool = False
    duplex: bool = False
    resolution: list[str] = []
    copies_supported: bool = False
    max_copies: Optional[int] = None
    page_ranges_supported: bool = False
    orientation_supported: bool = False


class PrinterSummary(BaseModel):
    name: str
    description: str
    state: str
    state_message: str
    accepting_jobs: bool
    shared: bool
    device_uri: Optional[str] = None
    current_job: Optional[int] = None
    queue_count: int = 0


class PrinterDetail(PrinterSummary):
    location: Optional[str] = None
    manufacturer: Optional[str] = None
    model: Optional[str] = None
    capabilities: PrinterCapabilities
